backend/de.metas.contracts/src/main/java/de/metas/contracts/commission/mediated/model/MediatedCommissionSettingsLineId.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * #%L
+ * de.metas.contracts
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.contracts.commission.mediated.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import de.metas.contracts.commission.model.I_C_MediatedCommissionSettingsLine;
+import de.metas.util.Check;
+import de.metas.util.lang.RepoIdAware;
+import lombok.NonNull;
+import lombok.Value;
+
+import javax.annotation.Nullable;
+
+@Value
+public class MediatedCommissionSettingsLineId implements RepoIdAware
```

Comment:
not a valid RepoIdAware class because a RepoIdAware shall wrap just an `int repoId`.
In your case, you are also wrapping the `MediatedCommissionSettingsId mediatedCommissionSettingsId`.

suggestion: don't implement RepoIdAware.

Reply:
I'd rather skip it from testing (see: `de.metas.util.lang.All_RepoIdAware_Classes_Test#skipRules`) like we do with other "composed" id classes... that way we maintain the advantages RepoIdAware offers when working with `InterfaceWrapperHelper`

Reply:
good point, agree